windows-agent/internal/cloudinit/cloudinit_test.go

```diff
@@ -36,12 +36,14 @@ func TestNew(t *testing.T) {
 				subcriptionErr: tc.breakWriteAgentData,
 			}
 
-			_, err := cloudinit.New(ctx, conf, publicDir)
+			ci, err := cloudinit.New(ctx, conf, publicDir)
 			if tc.wantErr {
 				require.Error(t, err, "Cloud-init creation should have returned an error")
+				require.Nil(t, ci, "Cloud-init creation should not have returned a CloudInit object")
```

Comment:
I would not do that one: it’s a convention you don’t care/trust the returned values if the last returned one is an error, which is not nil so let’s not add more assertions (or we should do that for everything else, in every special case).